packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -2951,6 +2951,116 @@ void main() {
     semantics.dispose();
   });
 
+  testWidgets('password fields hide the last char when obscure is ObscureTextBehavior.all', (WidgetTester tester) async {
+    final SemanticsTester semantics = SemanticsTester(tester);
+
+    controller.text = 'super-secret-password!!1';
+
+    await tester.pumpWidget(MaterialApp(
+      home: EditableText(
+        backgroundCursorColor: Colors.grey,
+        obscureTextBehavior: ObscureTextBehavior.all,
+        controller: controller,
+        focusNode: focusNode,
+        style: textStyle,
+        cursorColor: cursorColor,
+      ),
+    ));
+
+    final String expectedValue = '•' *controller.text.length;
+
+    expect(
+      semantics,
+      hasSemantics(
+        TestSemantics(
+          children: <TestSemantics>[
+            TestSemantics.rootChild(
+              children: <TestSemantics>[
+                TestSemantics(
+                  children: <TestSemantics>[
+                    TestSemantics(
+                      flags: <SemanticsFlag>[SemanticsFlag.scopesRoute],
+                      children: <TestSemantics>[
+                        TestSemantics(
+                          flags: <SemanticsFlag>[
+                            SemanticsFlag.isTextField,
+                            SemanticsFlag.isObscured,
+                          ],
+                          value: expectedValue,
+                          textDirection: TextDirection.ltr,
+                        ),
+                      ],
+                    )
+                  ],
+                )
+              ],
+            ),
+          ],
+        ),
+        ignoreTransform: true,
+        ignoreRect: true,
+        ignoreId: true,
+      ),
+    );
+
+    semantics.dispose();
+  });
+
+  testWidgets('password fields delayed hide the last char when obscure is ObscureTextBehavior.delayed', (WidgetTester tester) async {
+    final SemanticsTester semantics = SemanticsTester(tester);
+
+    controller.text = 'super-secret-password!!1';
+
+    await tester.pumpWidget(MaterialApp(
+      home: EditableText(
+        backgroundCursorColor: Colors.grey,
+        obscureTextBehavior: ObscureTextBehavior.delayed,
+        controller: controller,
+        focusNode: focusNode,
+        style: textStyle,
+        cursorColor: cursorColor,
+      ),
+    ));
+
+    final String expectedValue = '•' *controller.text.length;
```

Comment:
This doesn't test the delay, does it?  Maybe if you enter the text via `tester.enterText`, then check that the last character is not `"•"`, then wait a second and test that it has been obscured.